src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -0,0 +1,413 @@
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Pause state");
+
+        HandleCommand<Commands::Pause::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandlePauseState(ctx, req); });
+        break;
+
+    case Commands::Resume::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Resume state");
+
+        HandleCommand<Commands::Resume::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleResumeState(ctx, req); });
+        break;
+
+    case Commands::Start::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Start state");
+
+        HandleCommand<Commands::Start::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleStartState(ctx, req); });
+        break;
+
+    case Commands::Stop::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Stop state");
+
+        HandleCommand<Commands::Stop::DecodableType>(handlerContext,
+                                                     [this](HandlerContext & ctx, const auto & req) { HandleStopState(ctx, req); });
+        break;
+    }
+}
+
+CHIP_ERROR OperationalStateServer::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback,
+                                                             void * context)
+{
+    callback(Commands::Pause::Id, context);
+    callback(Commands::Resume::Id, context);
+    callback(Commands::Start::Id, context);
+    callback(Commands::Stop::Id, context);
+    return CHIP_NO_ERROR;
+}
+
+void OperationalStateServer::HandlePauseState(HandlerContext & ctx, const Commands::Pause::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "OperationalState: HandlePauseState");
+    Commands::OperationalCommandResponse::Type response;
+    OperationalErrorStateStruct currentErrState;
+    OperationalStateStruct currentOperationalState;
+
+    // copy current operational state
+    currentOperationalState = mOperationalState;
+    // copy current operational error
+    currentErrState = mOperationalError;
+
+    // callback
+    mDelegate->HandlePauseState(currentOperationalState, currentErrState);
+
+    // set operational error
+    SetOperationalState(currentOperationalState);
+
+    // copy current operational error
+    mOperationalError = currentErrState;
+
+    // return operational error
+    response.commandResponseState.errorStateID = static_cast<OperationalState::ErrorStateEnum>(mOperationalError.ErrorStateID);
+    response.commandResponseState.errorStateLabel.SetNonNull(mOperationalError.ErrorStateLabel,
+                                                             strlen(mOperationalError.ErrorStateLabel));
+    response.commandResponseState.errorStateDetails.Emplace(mOperationalError.ErrorStateDetails,
+                                                            strlen(mOperationalError.ErrorStateDetails));
```

Comment:
These are both wrong, for several reasons:

1) Applications must be able to send null for the label and not send a details.
2) This is not allowing applications to send the max spec-allowed string length.
3) None of the application API documents that these are null-terminated.